### ISSUE TITLE
Deprecation warnings for the wrong version of React in the new 7.0.0

### ISSUE THREAD

**Issue by almirfilho** [yannickcr/eslint-plugin-react#1182]
I'm getting the following deprecation warning, which is just fine:

```
warning  React.PropTypes is deprecated since React 15.5.0, use the npm module prop-types instead  react/no-deprecated
```

But, since I'm still using React __v15.4.2__ (and not 15.5.0+ as the message is pointing out), it doesn't seem to be the correct behavior. Am I right?
Thanks!

**Comment by ljharb** [comment]
Are you sure? Only React 15.5 prints that out. What does `npm ls` say?

**Comment by almirfilho** [comment]
Yes, I'm sure.

```
$ npm ls react

ukulele@1.0.3 /Users/almirfilho/Realgeeks/ukulele
└── react@15.4.2
```

It just began to show this warning after the upgrade of eslint-plugin-react v7.0.0.
I'm going migrate to the new proptypes anyway, it's just that I thought this could be a bug in the plugin.

**Comment by ljharb** [comment]
try `npm ls | grep react`?

**Comment by almirfilho** [comment]
```
npm ls | grep react

├─┬ eslint-plugin-react@7.0.0
├─┬ react@15.4.2
├─┬ react-addons-test-utils@15.4.2
├─┬ react-dom@15.4.2
```

**Comment by ljharb** [comment]
ahhh right, sorry. yes, that message is coming from this plugin.

I originally expected the deprecation warning would only print out on React 15.5 (per the version you specify in your config), but it seems it applies with all v15 versions.

@yannickcr, what do you think about restricting this check to 15.5+ only?

**Comment by yannickcr** [comment]
eslint-plugin-react can not detect which React version you are using, so it warns for every deprecated methods by default.

You can specify your React version in the [shared settings](https://github.com/yannickcr/eslint-plugin-react#configuration). This should solve your issue.

**Comment by ljharb** [comment]
@yannickcr right, but if I specify in settings that your version is "15.0" or "15.4" - as opposed to "15.5" - then i'd expect not to get this warning.

**Comment by yannickcr** [comment]
@ljharb Yes, and according to [these tests](https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/no-deprecated.js#L42-L45) it should work correctly. Were you able to reproduce the issue with the React version specified ?

**Comment by ljharb** [comment]
Ah, thanks for clarifying.

@almirfilho have you specified v15.4 in your settings?

**Comment by almirfilho** [comment]
@ljharb No, I haven't. I've just migrated to the new `prop-types`.
But thanks anyway!
